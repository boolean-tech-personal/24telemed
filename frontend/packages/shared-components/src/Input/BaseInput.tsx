import styled, { css } from 'styled-components';

export interface BaseInputProps {
  value?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  error: boolean;
  rightIcon?: React.ReactNode;
  leftIcon?: React.ReactNode;
  name: string;
  type: string;
  placeholder: string;
}

export function BaseInput({
  error,
  type,
  name,
  value,
  onChange,
  ...props
}: BaseInputProps) {
  return (
    <StyledInput
      {...props}
      onChange={onChange}
      value={value}
      error={error}
      type={type}
      aria-label={name}
    />
  );
}

const StyledInput = styled.input<
  Pick<BaseInputProps, 'error' | 'rightIcon' | 'leftIcon'>
>`
  padding-left: ${({ theme, leftIcon }) =>
    leftIcon ? theme.spacing.xl : theme.spacing.lg};
  padding-right: ${({ theme, rightIcon }) =>
    rightIcon ? theme.spacing.xl : theme.spacing.lg};
  padding-top: ${({ theme }) => theme.spacing.sm};
  padding-bottom: ${({ theme }) => theme.spacing.sm};
  border-radius: ${({ theme }) => theme.radius.textarea};
  font-size: ${({ theme }) => theme.typography.bodyMd.fontSize};

  ${({ theme }) =>
    theme.breakpoints.xs.down(css`
      font-size: ${({ theme }) => theme.typography.bodyMd.fontSize};
      width: 100%;
      height: ${({ theme }) => theme.spacing.md};
    `)}

  ${({ theme }) =>
    theme.breakpoints.xs.up(css`
      width: 100%;
      height: ${({ theme }) => theme.spacing.xl};
    `)}

  
  box-sizing: border-box;
  outline: none;
  ${({ error, theme }) => css`
    border: 1px solid
      ${error ? theme.palette.error : theme.palette.primary1.lighter};
  `}

  &:focus {
    border: 1px solid
      ${({ theme, error }) =>
        error ? theme.palette.error : theme.palette.secondary1.main};
  }
  font-family: ${({ theme }) => theme.typography.bodyMd.fontFamily};
  &::-webkit-input-placeholder {
    color: ${({ theme }) => theme.palette.primary1.lighter};
    font-size: ${({ theme }) => theme.typography.bodySm.fontSize};
    font-family: ${({ theme }) => theme.typography.bodyMd.fontFamily};
  }
`;
