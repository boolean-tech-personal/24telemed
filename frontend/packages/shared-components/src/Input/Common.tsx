import styled, { css } from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: ${({ theme }) => theme.spacing.md};
`;

export const HelperText = styled.span`
  ${({ theme }) =>
    theme.breakpoints.xs.down(css`
      font-size: ${({ theme }) => theme.typography.bodyXs.fontSize};
    `)}

  ${({ theme }) =>
    theme.breakpoints.xs.up(css`
      font-size: ${({ theme }) => theme.typography.bodySm.fontSize};
    `)}
  font-family: ${({ theme }) => theme.typography.bodyXs.fontFamily};
  color: #65a3ff;
`;

export const ErrorText = styled.span`
  font-size: ${({ theme }) => theme.typography.bodyXs.fontSize};
  font-family: ${({ theme }) => theme.typography.bodyXs.fontFamily};
  color: ${({ theme }) => theme.palette.error};
  margin-left: ${({ theme }) => theme.spacing.xs};
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const Label = styled.label`
font-weight: ${({ theme }) => theme.typography.fontWeight.bold};
font-family: ${({ theme }) => theme.typography.bodyXs.fontFamily};
  color: ${({ theme }) => theme.palette.primary1.main};
  ${({ theme }) =>
    theme.breakpoints.xs.down(css`
      font-size: ${({ theme }) => theme.typography.bodyXs.fontSize};
    `)}

    font-size: ${({ theme }) => theme.typography.bodySm.fontSize};
    
`;

export const RightIconContainer = styled.div`
  position: absolute;
  right: ${({ theme }) => theme.spacing.md};
  top: 50%;
  transform: translateY(-50%);
`;

export const InputContainer = styled.div`
  position: relative;
  
  display: inline-block;
`;

export const LeftIconContainer = styled.div`
  position: absolute;
  left: ${({ theme }) => theme.spacing.xs};
  top: 50%;
  transform: translateY(-50%);
`;
