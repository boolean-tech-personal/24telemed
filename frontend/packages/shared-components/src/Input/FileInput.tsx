import styled, { css } from 'styled-components';
import { AttachmentIcon, CancelIcon } from '../Icon';

interface FileInputProps {
  value: File | null;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onCancel: () => void;
  placeholder?: string;
}

export const FileInput = ({ value, onChange, onCancel, placeholder }: FileInputProps) => {
  return (
    <Container>
      <FileInputContainer>
        <AttachmentIcon />
        <Placeholder value={value}>{value ? value.name : placeholder}</Placeholder>

        <StyledFileInput
          type="file"
          onChange={onChange}
        />
      </FileInputContainer>
      <CancelIconContainer value={value} onClick={onCancel}>
        <CancelIcon />
      </CancelIconContainer>
    </Container>
  );
};

const StyledFileInput = styled.input`
  display: none;
`;

const FileInputContainer = styled.label`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spacing.sm};
  padding: ${({ theme }) => theme.spacing.sm};
  border: 1px solid ${({ theme }) => theme.palette.primary1.lighter};
  border-radius: ${({ theme }) => theme.spacing.xs};
  width: 368px;
  height: ${({ theme }) => theme.spacing.xs};
  cursor: pointer;
  &:hover {
    border-color: ${({ theme }) => theme.palette.secondary1.main};
  }

  ${({ theme }) =>
    theme.breakpoints.md.up(css`
      padding: ${({ theme }) => theme.spacing.md};
      width: 300px;
    `)}
`;

const Placeholder = styled.div<Pick<FileInputProps, 'value'>>`
  color: ${({ theme, value }) => value ? theme.palette.common.black : theme.palette.primary1.lighter};
  font-size: ${({ theme }) => theme.typography.bodySm.fontSize};
  line-height: ${({ theme }) => theme.typography.bodySm.lineHeight};
`;
const CancelIconContainer = styled.div<Pick<FileInputProps, 'value'>>`
  position: absolute;
  right: 24px;
  cursor: pointer;
  display: ${({ value }) => (value ? 'block' : 'none')};
  color: ${({ theme }) => theme.palette.neutral.light};
  top:70%;
  transform: translateY(-50%);
`;

const Container = styled.div`
  flex-direction: column;
  position: relative;
  width: fit-content;
  display: flex;
  height: 48px;
  padding: ${({ theme }) => theme.spacing.sm};
`;
