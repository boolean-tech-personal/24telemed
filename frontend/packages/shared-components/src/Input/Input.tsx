import styled from 'styled-components';
import { Controller } from 'react-hook-form';
import React from 'react';
import ErrorImage from '../assets/images/Frame.svg';
import HelperIcon from '../assets/images/HelperIcon.svg';
import AttachmentIcon from '../assets/images/AttachmentIcon.svg';
import CancelIcon from '../assets/images/CancelIcon.svg';
import Eye from '../assets/images/Eye.svg';

export interface InputProps {
  value: string;
  control: any;
  name: string;
  type: 'text' | 'file' | 'password'; 
  label: string;
  placeholder: string;
  inputRightIcon?: React.ReactNode;
  inputLeftIcon?: React.ReactNode;
  error?: string;
  helperText?: string;
}


export function Input(props: InputProps) {
  return (
    <InputLabelContainer>
      <StyledLabel>{props.label}</StyledLabel>
      <Container>
        {props.type === 'text' && (
          <>
            <Controller
              name={props.name}
              control={props.control}
              render={({ field: { onChange, value, ...field } }) => (
                <StyledInput
                  {...field}
                  type={props.type}
                  placeholder={props.placeholder}
                  onChange={onChange}
                  value={value}
                  error={props.error}
                />
              )}
            />
            {props.inputLeftIcon && <LeftIcon>{props.inputLeftIcon}</LeftIcon>}

            {props.inputRightIcon && (
              <RightIcon>{props.inputRightIcon}</RightIcon>
            )}
          </>
        )}

        {props.type === 'file' && (
          <FileLabel>
            <Controller
              name={props.name}
              control={props.control}
              render={({ field: { onChange, value, ...field } }) => (
                <FileInput
                  {...field}
                  type={props.type}
                  placeholder={props.placeholder}
                  onChange={onChange}
                  value={value}
                  error={props.error}
                />
              )}
            />

            <PlaceholderAttachContainer>
              <FileAttachIcon src={AttachmentIcon} alt="attachment" />
              <FilePlaceholder value={props.value}>
                Add attachment
              </FilePlaceholder>
            </PlaceholderAttachContainer>
          </FileLabel>
        )}
        {props.value && props.type === 'file' && (
          <CancelFileIcon
            src={CancelIcon}
            alt="cancel"
            onClick={() => props.control.setValue(props.name, '')}
          />
        )}

        {props.type === 'password' && (
          <>
            <Controller
              name={props.name}
              control={props.control}
              render={({ field: { onChange, value, ...field } }) => (
                <StyledInput
                  {...field}
                  type={props.type}
                  placeholder={props.placeholder}
                  onChange={onChange}
                  value={value}
                  error={props.error}
                />
              )}
            />

            <PasswordRightIcon
              src={Eye}
              alt="eye"
              onClick={() => props.control.setValue(props.name, '')}
            />
          </>
        )}
      </Container>

      {props.error && (
        <ErrorContainer>
          <ErrorImg src={ErrorImage} alt="error" />
          <ErrorText>{props.error}</ErrorText>
        </ErrorContainer>
      )}

      {props.helperText && (
        <HelperContainer>
          <HelperImg src={HelperIcon} alt="helper" />
          <HelperText>{props.helperText}</HelperText>
        </HelperContainer>
      )}
    </InputLabelContainer>
  );
}

const StyledInput = styled.input<Pick<InputProps, 'error'>>`
  padding-left: 40px;
  padding-right: 40px;
  padding-top: 10px;
  padding-bottom: 10px;
  border: 1px solid
    ${({ error, theme }) => (error ? theme.colors.error : theme.colors.primary)};
  border-radius: 10px;
  font-size: 16px;
  width: 368px;
  height: 48px;
  box-sizing: border-box;
  &:hover {
    border-color: ${({ error, theme }) =>
      error?.length ? theme.colors.error : theme.colors.secondary};
  }

  &:focus {
    outline: none;
    border-color: ${({ error, theme }) =>
      error ? theme.colors.error : theme.colors.tertiary};
  }
  & img {
    margin-right: 10px;
  }
`;

const StyledLabel = styled.label`
  font-size: ${(props) => props.theme.fontSizes.md};
  margin-bottom: 5px;
  display: block;
  font-family: ${(props) => props.theme.fonts.primary};
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
`;

const LeftIcon = styled.div`
  position: absolute;
  left: 10px;
  top: 50%;
  transform: translateY(-50%);
`;

const RightIcon = styled.div`
  position: absolute;
  right: 60px;
  top: 50%;
  transform: translateY(-50%);
`;

const InputLabelContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  width: 100%;
  max-width: 400px;
`;

const ErrorText = styled.span`
  color: ${(props) => props.theme.colors.error};
  font-size: ${(props) => props.theme.fontSizes.sm};
  padding-top: 5px;
  padding-bottom: 5px;
  line-height: 20px;
  font-family: ${({theme}) => theme.fonts.primary};
`;

const ErrorImg = styled.img`
  margin-right: 10px;
`;

const ErrorContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 20px;
`;

const HelperText = styled.span`
  color: ${(props) => props.theme.colors.helper};
  font-size: ${(props) => props.theme.fontSizes.sm};
  padding-top: 5px;
  padding-bottom: 5px;
  line-height: 20px;
  font-family: ${({theme}) => theme.fonts.primary};
`;

const HelperImg = styled.img`
  margin-right: 10px;
`;

const HelperContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 20px;
`;

const FileInput = styled.input<Pick<InputProps, 'error'>>`
  display: none;
`;

const FileLabel = styled.label`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 48px;
  background-color: #f5f5f5;
  border-radius: 10px;
  font-size: 16px;
  color: #000000;
  cursor: pointer;
  &:hover {
    background-color: #e5e5e5;
  }
  border: 1px solid ${(props) => props.theme.colors.primary};
`;

const FileAttachIcon = styled.img`
  margin-right: 10px;
  margin-left: 10px;
  width: 20px;
  height: 20px;
`;

const PlaceholderAttachContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
`;

const FilePlaceholder = styled.div<Pick<InputProps, 'value'>>`
  font-size: ${(props) => props.theme.fontSizes.sm};
  color: ${({ theme, value }) =>
    value ? theme.colors.value : theme.colors.primary};
  font-family: ${({theme}) => theme.fonts.primary};
`;

const CancelFileIcon = styled.img`
  margin-right: 10px;
  width: 20px;
  height: 20px;
  position: absolute;
  right: 10px;
  top: 50%;
  transform: translateY(-50%);

  &:hover {
    cursor: pointer;
  }
`;

const PasswordRightIcon = styled.img`
  margin-right: 10px;
  width: 20px;
  height: 20px;
  position: absolute;
  right: 50px;
  top: 50%;
  transform: translateY(-50%);
  &:hover {
    cursor: pointer;
  }
`;
