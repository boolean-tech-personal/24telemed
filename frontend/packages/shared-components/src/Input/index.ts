export * from './BaseInput'
export * from './TextInput'
export * from './PasswordInput'
export * from './FileInput'
export * from './NumberInput'