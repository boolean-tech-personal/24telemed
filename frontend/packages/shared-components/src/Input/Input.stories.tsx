import type { Meta, StoryObj, StoryFn } from '@storybook/react';
import { Input, InputProps } from './Input';
import { ThemeProvider } from 'styled-components';
import { theme } from '../config';
import { useForm } from 'react-hook-form';
import * as yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup';
import { FormProvider } from 'react-hook-form';
import { FaBeer } from 'react-icons/fa';

type Story = StoryObj<typeof Input>;

type FormDataType = {
  input: string;
};

const Template: StoryFn<InputProps> = (args) => {
  const schema = yup.object().shape({
    input: yup.string().required(),
  });

  const method = useForm<FormDataType>({
    resolver: yupResolver(schema),
    defaultValues: {
      input: '',
    },
  });

  console.log(method.watch('input'));

  return <Input {...args} control={method.control} value= {method.watch('input')} />;
};

const meta: Meta<typeof Input> = {
  component: Input,
  title: 'Input',
  decorators: [
    (Story) => (
      <ThemeProvider theme={theme}>
        <Story />
      </ThemeProvider>
    ),
  ],
};

export const Default = Template.bind({});
Default.args = {
  name: 'input',
  label: 'Field Label',
  placeholder: 'Type here...',
  type: 'text',
};

export default meta;

export const InputTextWithoutLabel = Template.bind({});

InputTextWithoutLabel.args = {
  name: 'input',
  placeholder: 'Type here...',
  type: 'text',
  value: '',
};

export const InputTextwithRightIcon = Template.bind({});

InputTextwithRightIcon.args = {
  name: 'input',
  label: 'Field Label',
  placeholder: 'Type here...',
  type: 'text',
  value: '',
  inputRightIcon: <FaBeer />,
};

export const InputTextwithLeftIcon = Template.bind({});
InputTextwithLeftIcon.args = {
  name: 'input',
  label: 'Field Label',
  placeholder: 'Type here...',
  type: 'text',
  value: '',
  inputLeftIcon: <FaBeer />,
};

export const InputTextWithHelperText = Template.bind({});
InputTextWithHelperText.args = {
  name: 'input',
  label: 'Field Label',
  placeholder: 'Type here...',
  type: 'text',
  value: '',
  helperText: 'This is a helper text',
};

export const InputTextWithError = Template.bind({});

InputTextWithError.args = {
  name: 'input',
  label: 'Field Label',
  placeholder: 'Type here...',
  type: 'text',
  value: '',
  error: 'This is an error',
};

export const InputFileWithLabel = Template.bind({});

InputFileWithLabel.args = {
  name: 'input',
  label: 'Upload file',
  type: 'file',
  value: '',
  placeholder: 'Add attachment...',
};

export const InputFileWithHelperText = Template.bind({});

InputFileWithHelperText.args = {
  name: 'input',
  label: 'Upload file',
  type: 'file',
  value: '',
  placeholder: 'Add attachment...',
  helperText: 'This is a helper text',
};

export const InputFileWithError = Template.bind({});

InputFileWithError.args = {
  name: 'input',
  label: 'Upload file',
  type: 'file',
  value: '',
  placeholder: 'Add attachment...',
  error: 'This is an error',
};

export const InputFileWithValue = Template.bind({});

InputFileWithValue.args = {
  name: 'input',
  label: 'Upload file',
  type: 'file',
  value: 'file.txt',
  placeholder: 'Add attachment...',
};

export const PasswordInput = Template.bind({});

PasswordInput.args = {
  name: 'input',
  label: 'Password',
  type: 'password',
  value: '',
  placeholder: 'Type here...',
};
