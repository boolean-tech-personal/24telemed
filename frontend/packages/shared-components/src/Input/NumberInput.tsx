import { ErrorIcon, HelperIcon, StarIcon, NigeriaFlagIcon } from '../Icon';

import {
  Container,
  Label,
  InputContainer,
  TextContainer,
  HelperText,
  ErrorText,
} from './Common';
import {BaseInput} from './BaseInput';
import styled from 'styled-components';
export interface NumberInputProps extends React.ComponentProps<'input'> {
  helperText?: string;
  error: boolean;
  errorText?: string;
  label?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value: string;
  leftIcon?: React.ReactNode;
}

export function NumberInput({
  helperText,
  error,
  errorText,
  label,
  value,
  onChange,
  leftIcon,
  
  ...rest
}: NumberInputProps) {
  return (
    <Container>
      {label && <Label>{label}</Label>}
      <InputContainer>
        {
          leftIcon && <IconContainer>{leftIcon}</IconContainer>
          
        }
        <BaseInput {...rest} type='tel' name="numberInput" value={value} onChange={onChange} error={error} placeholder="+234"/>
      </InputContainer>
      {helperText && (
        <TextContainer>
          <HelperIcon />
          <HelperText>{helperText}</HelperText>
        </TextContainer>
      )}
      {error && errorText && (
        <TextContainer>
          <ErrorIcon />
          <ErrorText>{errorText}</ErrorText>
        </TextContainer>
      )}
    </Container>
  );
}

const IconContainer = styled.div`
  position: absolute;
  left: ${({ theme }) => theme.spacing.xs};
  top: 50%;
  transform: translateY(-50%);
`;
