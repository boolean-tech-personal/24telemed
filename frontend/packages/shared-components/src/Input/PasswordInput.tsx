import styled from 'styled-components';
import { BaseInput } from './BaseInput';
import { useState } from 'react';

import {
  Container,
  Label,
  InputContainer,
  TextContainer,
  HelperText,
  ErrorText,
} from './Common';
import { EyeIcon, HelperIcon,ErrorIcon } from '../Icon';

export interface PasswordInputProps
  extends React.ComponentProps<typeof BaseInput> {
  helperText?: string;
  error: boolean;
  errorText?: string;
  label?: string;
}

export function PasswordInput({
  helperText,
  error,
  errorText,
  label,
  ...rest
}: PasswordInputProps) {
  const [textType, setTextType] = useState('password');
  const toggleShowPassword = () => {
    setTextType(textType === 'password' ? 'text' : 'password');
  };
  return (
    <Container>
      {label && <Label>{label}</Label>}
      <InputContainer>
        <BaseInput {...rest} type={textType} error={error} />

        <EyeIconContainer onClick={toggleShowPassword}>
          <EyeIcon />
        </EyeIconContainer>
      </InputContainer>
      {helperText && (
        <TextContainer>
          <HelperIcon />
          <HelperText>{helperText}</HelperText>
        </TextContainer>
      )}
      {error && errorText && (
        <TextContainer>
          <ErrorIcon />
          
          <ErrorText>{errorText}</ErrorText>
        </TextContainer>
      )}
    </Container>
  );
}

const EyeIconContainer = styled.div`
  position: absolute;
  right: ${({ theme }) => theme.spacing.md};
  top: 50%;
  transform: translateY(-50%);
  cursor: pointer;
`;
