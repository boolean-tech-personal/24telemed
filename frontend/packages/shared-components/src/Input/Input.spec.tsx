import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, renderHook } from '@testing-library/react';
import { Input } from './Input';
import { useForm } from 'react-hook-form';
import { ThemeProvider } from 'styled-components';
import { theme } from '../config';
type FormDataType = {
  input: string;
};
describe('Input', () => {
  const { result } = renderHook(() => useForm<FormDataType>({
    defaultValues: {
      input: '',
    },
  }));

  it('renders an input', () => {
    render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="text"
          value=""
        />
      </ThemeProvider>
    );
    const input = screen.getByPlaceholderText('Type here...');
    expect(input).toBeInTheDocument();
  });

  it('renders a label', () => {
    render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="text"
          value=""
        />
      </ThemeProvider>
    );
    const label = screen.getByText('Field Label');
    expect(label).toBeInTheDocument();
  });

  it('renders a file input', () => {
    render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="file"
          value=""
        />
      </ThemeProvider>
    );
    const input = screen.getByPlaceholderText('Type here...');
    expect(input).toBeInTheDocument();
  });

  it('renders a file label', () => {
    render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="file"
          value=""
        />
      </ThemeProvider>
    );
    const label = screen.getByText('Field Label');
    expect(label).toBeInTheDocument();
  });

  it('renders a file input with a file icon', () => {
    render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="file"
          value=""
        />
      </ThemeProvider>
    );
    const icon = screen.getByAltText('attachment');
    expect(icon).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Input
          control={result.current.control}
          name="input"
          label="Field Label"
          placeholder="Type here..."
          type="text"
          value=""
        />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  }
  );

  
});
