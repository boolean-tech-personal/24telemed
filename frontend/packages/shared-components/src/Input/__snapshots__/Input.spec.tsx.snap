// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jSUZER jOkPEt"
  >
    <label
      class="sc-gswNZR oFCHO"
    >
      Field Label
    </label>
    <div
      class="sc-dkrFOg OaQBh"
    >
      <input
        class="sc-bcXHqe fVFQcv"
        name="input"
        placeholder="Type here..."
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
